pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2107,7 +2141,7 @@ func buildListener(port api.ServicePort, annotations map[string]string) (*elb.Li
 	listener.InstancePort = &instancePort
 	listener.LoadBalancerPort = &loadBalancerPort
 	certID := annotations[ServiceAnnotationLoadBalancerCertificate]
-	if certID != "" {
+	if certID != "" && (sslPorts == nil || sslPorts.numbers.Has(loadBalancerPort) || sslPorts.names.Has(portName)) {
```

Comment:
Nit: we could probably move this to a method on portSets (and also avoid the nil check by returning a special value), and also pass in api.ServicePort.  But probably for 1.4...